### ISSUE TITLE
use quiet

### ISSUE THREAD

**Issue by FernandoMiguel** [actions/checkout#18]
currently the checkout performs:
`git -c http.extraheader="AUTHORIZATION: basic ***" fetch --tags --prune --progress --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/*`

would be great to use quiet instead

`git -c http.extraheader="AUTHORIZATION: basic ***" fetch --tags --prune --quiet --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/*`

so the job log is cleaner to read

**Comment by TingluoHuang** [comment]
We can provide an option to turn on `--quite`
There are 2 reasons we have `--progress`
- When you watch the workflow running, you won't feel the `git fetch` is hanging.
- When you try to investigate why my `checkout` action is slow, the progress output can give some idea about how many objects you are fetching and what's the average object download speed.

**Comment by FernandoMiguel** [comment]
Didn't know the command option was available.
Maybe add it to the docs? 
Thanks

**Comment by TingluoHuang** [comment]
@FernandoMiguel we don't have it right now, we can consider adding it. 😄

**Comment by FernandoMiguel** [comment]
The docs or the flag? 🤔

**Comment by TingluoHuang** [comment]
the flag. 😄

**Comment by FernandoMiguel** [comment]
Damn it...
That explains why my build failed lol
Git revert Ahah

**Comment by FernandoMiguel** [comment]
This is what I get for half reading comments while having breakfast.
Cheers

**Comment by moltar** [comment]
Agree with OP, the log seems to be quite verbose:

```

Receiving objects:   6% (19/304)

                                                                                
Receiving objects:   7% (22/304)

                                                                                
Receiving objects:   8% (25/304)

                                                                                
Receiving objects:   9% (28/304)

                                                                                
Receiving objects:  10% (31/304)

                                                                                
Receiving objects:  11% (34/304)

                                                                                
Receiving objects:  12% (37/304)

                                                                                
Receiving objects:  13% (40/304)

                                                                                
Receiving objects:  14% (43/304)

                                                                                
Receiving objects:  15% (46/304)
```

But I can also see it being useful. 

I wish there was a way to disable that though. I am not really watching checkout ever.

**Comment by wipe2238** [comment]
>When you watch the workflow running, you won't feel the git fetch is hanging.

Problem is, Actions tab really have problems with handling long and quickly updating steps, up to the point that it's freezing browser while action is running. That's [UI issue](https://github.community/t5/GitHub-Actions/Slow-console-log-rendering/m-p/33200#M1407), not anything wrong with action itself, but `actions/checkout` is usually running early, and is used in most workflows around.
So i can't watch workflow running anyway, unless i open tab only _after_ checkout finishes and live log is hidden :/

TL;DR i'd like having `quiet` here, just to have live preview actually working; frontend team failed to deliver, so maybe backend team can make it happen ;)

**Comment by ericsciple** [comment]
Marking as enhancement and I'll defer to others.

Personally, +1 what @TingluoHuang said. Sense of motion and timestamps are good for us to help customers. I'm generally hesitant to add extra inputs (overload user with options).

**Comment by sauyon** [comment]
I feel like the real solution here would be to add support for the carriage return clearing the line.

**Comment by ericsciple** [comment]
Output from the step is now [grouped](https://github.com/actions/checkout/pull/191) so that should help.

**Comment by timotheecour** [comment]
group doesn't help much if you're reading raw logs... can we please have `--quiet` option?

**Comment by grosser** [comment]
@ericsciple plz reopen, it's still wasteful/unhelpful to have all these extra log lines
quiet as default and a flag to undo would be great

**Comment by kokosing** [comment]
For big project like https://github.com/trinodb/trino where we have plenty of jobs (currently 35) we we run very often (like 40 times per day). We are pretty sure that is working just fine as it has never failed for us (at least I don't recall a case), but still every single log file has about 170 lines with git progress that does not add any value. We usually read the raw log file as UI is not responsive with such huge log files (browser is not much responsive too but it is a bit better with raw files).

Maybe we could have `debug` option to troubleshoot problems with there are issues with checkout instead? And have `--quite` to be used by default?

**Comment by Allon-Guralnek** [comment]
The amount of logs this generates is insane. Please consider at least giving us the option to quiet it down.

**Comment by rgoliveira** [comment]
Just to add another POV: depending on how many jobs/runs and the nature of them, a repo can quickly use large amounts of storage space to accommodate these logs. I fail to see how the proposed flag could be a problem.

**Comment by froger-me** [comment]
Please reopen

**Comment by simonbaird** [comment]
FYI I think I have a good solution for this, see #1067 .
